extension-workspace-references.md

```diff
@@ -117,3 +117,26 @@ interface SymbolLocationInformation {
 }
 ```
 * error: code and message set in case an exception happens during the definition request.
+
+
+### Extended Workspace Symbol Request
+
+The `workspace/symbol` request takes an optional parameter `symbol` that allows you to query by known properties about the symbol.
+The string `query` parameter becomes optional.
+
+```typescript
+/**
+ * The parameters of a Workspace Symbol Request.
+ */
+interface WorkspaceSymbolParams {
+    /**
+     * A query string
+     */
+    query?: string;
+
+    /**
+     * Known properties about the symbol.
+     */
+    symbol?: Partial<SymbolDescriptor>;
```

Comment:
1. I like to keep it short and precise - if any, I would prefer `symbolQuery`, and that is kind of confusing since the whole thing you are doing is a symbol query. I thought about naming it `symbolDescriptor`, but it is not a full `SymbolDescriptor`, and I would like to change it to `SymbolInformation` soon.
2. Yes, this should be specified. The semantics should be an AND, because it could be useful to query with a fuzzy `query` but restrict it to a specific `SymbolKind` or `package`.